packages/angular_devkit/build_angular/src/builders/jest/index.ts

```diff
@@ -0,0 +1,147 @@
+        error:
+          'Jest is not installed, most likely you need to run `npm install jest --save-dev` in your project.',
+      };
+    }
+
+    // Verify that JSDom is installed in the project.
+    const environment = require.resolve('jest-environment-jsdom');
+    if (!environment) {
+      return {
+        success: false,
+        error:
+          '`jest-environment-jsdom` is not installed. Install it with `npm install jest-environment-jsdom --save-dev`.',
+      };
+    }
+
+    // Build all the test files.
+    const testFiles = await findTestFiles(options, context.workspaceRoot);
+    const jestGlobal = path.join(__dirname, 'jest-global.mjs');
+    const initTestBed = path.join(__dirname, 'init-test-bed.mjs');
+    const buildResult = await build(context, {
+      // Build all the test files and also the `jest-global` and `init-test-bed` scripts.
+      entryPoints: new Set(Array.from(testFiles).concat([jestGlobal, initTestBed])),
+      tsConfig: options.tsConfig,
+      polyfills: options.polyfills ?? ['zone.js', 'zone.js/testing'],
+      outputPath: testOut,
+      aot: false,
+      index: null,
+      outputHashing: OutputHashing.None,
+      outExtension: 'mjs', // Force native ESM.
+      commonChunk: false,
+      optimization: false,
+      buildOptimizer: false,
+      sourceMap: {
+        scripts: true,
+        styles: false,
+        vendor: true,
+      },
+    });
+    if (!buildResult.success) {
+      return buildResult;
+    }
+
+    // Execute Jest on the built output directory.
+    const jestProc = execFile('node', [
+      '--experimental-vm-modules',
+      jest,
+
+      `--rootDir=${testOut}`,
+      '--testEnvironment=jsdom',
+      '--color',
+
+      // Run basically all files in the output directory, any excluded files were already dropped by the build.
+      `--testMatch=${path.join('<rootDir>', '**', '*.mjs')}`,
+
+      // Load polyfills and initialize the environment before executing each test file.
+      // IMPORTANT: Order matters here.
+      // First, we execute `jest-global.mjs` to initialize the `jest` global variable.
+      // Second, we execute user polyfills, including `zone.js` and `zone.js/testing`. This is dependent on the Jest global so it can patch
+      // the environment for fake async to work correctly.
+      // Third, we initialize `TestBed`. This is dependent on fake async being set up correctly beforehand.
+      `--setupFilesAfterEnv=${path.join('<rootDir>', 'jest-global.mjs')}`,
+      `--setupFilesAfterEnv=${path.join('<rootDir>', 'polyfills.mjs')}`,
+      `--setupFilesAfterEnv=${path.join('<rootDir>', 'init-test-bed.mjs')}`,
+
+      // Don't run any infrastructure files as tests, they are manually loaded where needed.
+      `--testPathIgnorePatterns=${path.join('<rootDir>', 'jest-global\\.mjs')}`,
+      `--testPathIgnorePatterns=${path.join('<rootDir>', 'polyfills\\.mjs')}`,
+      `--testPathIgnorePatterns=${path.join('<rootDir>', 'init-test-bed\\.mjs')}`,
```

Comment:
The `path.join`'s here are redundant. Also, there is no need to create a regexp pattern for `testPathIgnorePatterns` as these can simple be paths.

I also suggest that values are wrapped in quotes to avoid possible issues when there are special characters or spaces in the paths. 

```suggestion
      // Third, we initialize `TestBed`. This is dependent on fake async being set up correctly beforehand.
      `--setupFilesAfterEnv='<rootDir>/jest-global.mjs'`,
      `--setupFilesAfterEnv='<rootDir>/polyfills.mjs'`,
      `--setupFilesAfterEnv='<rootDir>/init-test-bed.mjs'`,

      // Don't run any infrastructure files as tests, they are manually loaded where needed.
      `--testPathIgnorePatterns='<rootDir>/jest-global.mjs'`,
      `--testPathIgnorePatterns='<rootDir>/polyfills.mjs'`,
      `--testPathIgnorePatterns='<rootDir>/init-test-bed.mjs'`,
```

Reply:
I had a feeling those `path.join()`'s were unnecessary, but wasn't too sure if Windows would throw a wrench into it. `--testPathIgnorePatterns` is treated as a regex/glob, so the `\.` is needed to be specific about the file. https://jestjs.io/docs/cli#--testpathignorepatternsregexarray

The quotes is a good point. I think we _should_ be skipping shell expansion here, but good to be safe.

Reply:
Based on the tests https://github.com/jestjs/jest/blob/268afca708199c0e64ef26f35995907faf4454ff/e2e/multi-project-config-root/package.json the values can be string paths.